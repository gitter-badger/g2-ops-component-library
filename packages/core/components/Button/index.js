export { IconButton, Button } from './Button'
