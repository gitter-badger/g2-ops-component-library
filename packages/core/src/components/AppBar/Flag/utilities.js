import { COUNTRY_CODES, FLAG_URL_PREFIX, HEIGHT_22PX, HEIGHT_30PX } from './consts'

const code = (countryCode: string): string => {
    return COUNTRY_CODES[countryCode] || countryCode
}
  
export const flagUrl = (countryCode: string): string => {
    return `${FLAG_URL_PREFIX}${code(countryCode)}.png`
}

export const imageClass = (type: string) => {
    return type === 'cas'
        ? HEIGHT_22PX
        : HEIGHT_30PX;
}