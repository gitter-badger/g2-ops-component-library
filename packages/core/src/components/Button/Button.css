.copartButton {
	height: 36px;
	min-width: 110px !important;
}

.copartButton button:disabled {
	cursor: not-allowed !important;
}

.copartButton button span {
	font-weight: 500 !important;
	text-transform: capitalize !important;
}

.primaryButton button {
	background-color: #1d5ab9 !important;
}

.primaryButton button:hover {
	background-color: #133a77 !important;
}

.primaryButton button span {
	color: white !important;
}

.secondaryButton button {
	background-color: white !important;
}

.secondaryButton button:hover {
	background-color: #d9d9d9 !important;
}

.inactiveButton button {
	background-color: #dfdfdf !important;
}

.inactiveButton button:hover {
	background-color: #b9b9b9 !important;
}

.deleteButton button {
	background-color: #ee2727 !important;
}

.deleteButton button:hover {
	background-color: #ba0f0f !important;
}

.deleteButton button span {
	color: white !important;
}

.addButton button svg, .editButton button svg, .deleteButton button svg {
	margin: 0 !important;
	color: white !important;
	fill: white !important;
}

.addButton button span, .editButton button span, .deleteButton button span {
	padding-left: 3px !important;
}

.addButton button svg, .deleteButton button svg {
	height: 20px !important;
	width: 20px !important;
}

.editButton button svg {
	height: 16px !important;
	width: 16px !important;
}