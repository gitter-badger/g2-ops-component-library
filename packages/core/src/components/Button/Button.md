Button Component Examples:
```jsx
<div>
  <span style={{ margin: '10px' }}><Button type="primary" label="Copart Primary Button" /></span>
  <span style={{ margin: '10px' }}><Button type="secondary" label="Secondary Button" /></span>
  <span style={{ margin: '10px' }}><Button type="inactive" disabled label="Inactive" /></span>
  <span style={{ margin: '10px' }}><Button type="add" /></span>
  <span style={{ margin: '10px' }}><Button type="edit" /></span>
  <span style={{ margin: '10px' }}><Button type="delete" /></span>
</div>
```

Icon Button Examples:

Render any icon from material icons that can be found here: https://material.io/icons/
```js
const { IconButton } = require('./Button');
const iconsList = [
  'save', 'delete_forever', 'close',
  'assignment', 'autorenew', 'cached',
  'call', 'cancel', 'create', 'exit_to_app',
  'favorite', 'filter_list', 'forward',
  'help', 'lock_outline', 'offline_pin',
  'open_in_new', 'phone', 'print', 'receipt',
  'replay', 'search', 'settings_backup_restore',
  'settings', 'sort', 'swap_horiz', 'add_circle',
  'remove_circle'
];
<div>
  {iconsList.map(iconName => (
    <IconButton
      key={iconName}
      tooltip={iconName}
      onClick={() => console.log(`${iconName} clicked.`)}>
      <i className="material-icons">{iconName}</i>
    </IconButton>
  ))}
</div>
```