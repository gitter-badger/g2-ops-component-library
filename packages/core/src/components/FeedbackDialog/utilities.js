import { filter, isNil } from 'ramda'

import { SUBJECT_HEADER } from './consts';

export const createSubjectText = (target) => {
	return `${SUBJECT_HEADER} Feedback - Process: ${target.process} - Issue Type: ${target.issueType}`
}

// NOTE: Cloned from ops-portal repo.
export const generateFeedbackEmail = (props) => {
  return `
    Feedback: ${props.feedback}
    <br/>
    Email: ${props.email}
    <br/>
    Url: ${location.href}
    <br/>
    Selected Yard: ${props.selectedYard}
    <br/>
    Home Yard: ${props.homeYard}
    <br/>
    User Agent: ${navigator.userAgent}
  `
}

// NOTE: Cloned from ops-portal repo.
export const copyToClipboard = (text) => {
  const textField = document.createElement('textarea')
  textField.innerText = text
  document.body.appendChild(textField)
  textField.select()
  document.execCommand('copy')
  textField.remove()
}

// TODO: Clean up.
export const toObject = (form) => {
  if (!form || form.nodeName !== 'FORM') {
    return {}
  }
  const obj = {}
  let j = 0
  for (let i = form.elements.length - 1; i >= 0; i--) {
    if (form.elements[i].name === '') {
      continue
    }
    switch (form.elements[i].nodeName) {
      case 'INPUT':
        switch (form.elements[i].type) {
          case 'text':
          case 'email':
          case 'hidden':
          case 'password':
          case 'button':
          case 'reset':
          case 'submit':
            obj[form.elements[i].name] = form.elements[i].value
            break
          case 'checkbox':
          case 'radio':
            if (form.elements[i].checked) {
              obj[form.elements[i].name] = form.elements[i].value
            }
            break
          default:
            break
        }
        break
      case 'TEXTAREA':
        obj[form.elements[i].name] = form.elements[i].value
        break
      case 'SELECT':
        switch (form.elements[i].type) {
          case 'select-one':
            obj[form.elements[i].name] = form.elements[i].value
            break
          case 'select-multiple':
            for (j = form.elements[i].options.length - 1; j >= 0; j -= 1) {
              if (form.elements[i].options[j].selected) {
                obj[form.elements[i].name] = form.elements[i].options[j].value
              }
            }
            break
          default:
            break
        }
        break
      case 'BUTTON':
        switch (form.elements[i].type) {
          case 'reset':
          case 'submit':
          case 'button':
            obj[form.elements[i].name] = form.elements[i].value
            break
          default:
            break
        }
        break
      default:
        break
    }
  }
  return obj
}

// TODO: Clean up.
export const objectToQuery = (obj) => {
  const parts = []
  for (const i in obj) {
    if (obj.hasOwnProperty(i)) {
      if (Array.isArray(obj[i])) {
        [...new Set(obj[i])].map((val) => parts.push(`${encodeURIComponent(i)}=${encodeURIComponent(val)}`))
      } else {
        parts.push(`${encodeURIComponent(i)}=${encodeURIComponent(obj[i])}`)
      }
    }
  }
  return parts.join('&')
}

export const serialize = (form) => {
  const obj = toObject(form)
  return objectToQuery(obj)
}

const isClean = (x) => !(isNil(x) || x === '')
export const cleanData = (data) => filter(isClean, data)

// TODO: Clean up.
export const removeEmpty = (obj) => {
  const nonEmptyObj = {}
  for (const i in obj) {
    if (obj.hasOwnProperty(i) && obj[i] && !(obj[i] === 'A')) {
      nonEmptyObj[i] = obj[i]
    }
  }
  return nonEmptyObj
}

// TODO: Clean up.
export const dotToObj = (obj) => {
  const newObj = {}
  for (const k in obj) {
    if (obj.hasOwnProperty(k)) {
      const parts = k.split('.')
      const parentKey = parts.shift()
      newObj[parentKey] = parts.reverse().reduce((o, i) => {
        const res = {}
        res[i] = o
        return res
      }, obj[k])
    }
  }
  return newObj
}