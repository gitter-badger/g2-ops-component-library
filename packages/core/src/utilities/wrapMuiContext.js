// @flow
import type { ComponentType, Node } from 'react'

import React, { Component } from 'react'
import injectTapEventPlugin from 'react-tap-event-plugin'
import MuiThemeProvider from 'material-ui/styles/MuiThemeProvider'
import getMuiTheme from 'material-ui/styles/getMuiTheme'
import { initializeIcons } from '@uifabric/icons'

type ThemeType = {
  [string]: any,
}

const muiTheme = getMuiTheme()
const copartBlue = '#1d5ab9'
const niceBlue = '#4a90e2'

export const copartBaseTheme: ThemeType = {
    ...muiTheme,
  appBar: {
    ...muiTheme.appBar,
    color: copartBlue,
  },
  baseTheme: {
    ...muiTheme.baseTheme,
    palette: {
      ...muiTheme.baseTheme.palette,
      pickerHeaderColor: copartBlue,
      primary1Color: copartBlue,
      primary2Color: copartBlue,
    },
  },
  flatButton: {
    ...muiTheme.flatButton,
    primaryTextColor: copartBlue,
  },
  palette: {
    ...muiTheme.palette,
    pickerHeaderColor: copartBlue,
    primary1Color: copartBlue,
    primary2Color: copartBlue,
  },
  datePicker: {
    ...muiTheme.datePicker,
    headerColor: copartBlue,
    color: copartBlue,
    selectColor: copartBlue,
  },
  toggle: {
    ...muiTheme.toggle,
    thumbOnColor: copartBlue,
    trackOnColor: niceBlue,
  },
  timePicker: {
    ...muiTheme.timePicker,
    headerColor: copartBlue,
    accentColor: copartBlue,
    selectColor: copartBlue,
  },
  stepper: {
    ...muiTheme.stepper,
    iconColor: copartBlue,
  },
  tabs: {
    selectedTextColor: 'rgba(38, 166, 91, 1.0)',
    textColor: 'rgba(218, 223, 225, 1.0)',
  },
  inkBar: {
    backgroundColor: '#545A63',
  },
}

initializeIcons()
injectTapEventPlugin()

// a higher order function that provides mui context to the Component being passed.
export function wrapMuiContext<T>(WrapperComponent: ComponentType<T>): (T) => Node {
  return (props) => (
    <MuiThemeProvider muiTheme={copartBaseTheme}>
      <WrapperComponent {...props} />
    </MuiThemeProvider>
  )
}
