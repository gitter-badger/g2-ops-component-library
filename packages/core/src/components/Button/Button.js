// @flow
import React from 'react'
import RaisedButton from 'material-ui/RaisedButton'
import MuiIconButton from 'material-ui/IconButton'
import AddIcon from 'material-ui/svg-icons/content/add'
import EditIcon from 'material-ui/svg-icons/image/edit'
import DeleteIcon from 'material-ui/svg-icons/action/delete-forever'

import { wrapMuiContext } from 'utilities/wrapMuiContext'

// TODO: Fix button + icon styling.
import './Button.css'

type ButtonPropsT = {
  /** To indicate type of button {primary, secondary, inactive, add, edit, delete} */
  variant: 'primary' | 'secondary' | 'inactive' | 'add' | 'edit' | 'delete',
  /** OnClick event for button */
  onClick: (SyntheticMouseEvent<HTMLInputElement>) => void,
}

export const Button = wrapMuiContext(({ variant, ...buttonProps }: ButtonPropsT) => {
	let typeProps = {}

  switch (variant) {
    case 'primary':
    case 'secondary':
    case 'inactive':
      typeProps = {
        ...buttonProps,
        className: `copartButton ${variant}Button ${buttonProps.className}`,
      }
      break


    case 'add':
      typeProps = {
        ...buttonProps,
        label: 'Add',
        className: `copartButton primaryButton ${variant}Button ${buttonProps.className}`,
        icon: <AddIcon />,
      }
      break

    case 'edit':
      typeProps = {
        ...buttonProps,
        label: 'Edit',
        className: `copartButton primaryButton ${variant}Button ${buttonProps.className}`,
        icon: <EditIcon />,
      }
      break

    case 'delete':
      typeProps = {
        ...buttonProps,
        label: 'Delete',
        className: `copartButton ${variant}Button ${buttonProps.className}`,
        icon: <DeleteIcon />,
      }
      break
    
    default:
      typeProps = { ...buttonProps }

  }

  return <RaisedButton {...typeProps} data-core-component="Button" />
})

Button.defaultProps = {
	className: ''
}

export const IconButton = wrapMuiContext(MuiIconButton)


