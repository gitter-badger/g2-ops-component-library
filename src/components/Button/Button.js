import React from 'react'
import PropTypes from 'prop-types'
import RaisedButton from 'material-ui/RaisedButton'
import MuiIconButton from 'material-ui/IconButton'
import AddIcon from 'material-ui/svg-icons/content/add'
import EditIcon from 'material-ui/svg-icons/image/edit'
import DeleteIcon from 'material-ui/svg-icons/action/delete-forever'
import { wrapMuiContext } from '../../wrapMuiContext'
import './style.scss'

const buttonPropTypes = {
  type: PropTypes.oneOf([
    'primary',
    'secondary',
    'inactive',
    'add',
    'edit',
    'delete'
  ]).isRequired
}
const Button = ({ primary, secondary, type, ...buttonProps }) => {
  let typeProps
  switch (type) {
    case 'primary':
    case 'secondary':
    case 'inactive': {
      typeProps = {
        ...buttonProps,
        className: `copartButton ${type}Button`
      }
      break
    }
    case 'add': {
      typeProps = {
        ...buttonProps,
        label: 'Add',
        className: `copartButton primaryButton ${type}Button`,
        icon: <AddIcon />
      }
      break
    }
    case 'edit': {
      typeProps = {
        ...buttonProps,
        label: 'Edit',
        className: `copartButton primaryButton ${type}Button`,
        icon: <EditIcon />
      }
      break
    }
    case 'delete': {
      typeProps = {
        ...buttonProps,
        label: 'Delete',
        className: `copartButton ${type}Button`,
        icon: <DeleteIcon />
      }
      break
    }
    default: {
      typeProps = buttonProps
    }
  }
  return <RaisedButton {...typeProps} />
}

Button.propTypes = buttonPropTypes

export const IconButton = wrapMuiContext(MuiIconButton)
export default wrapMuiContext(Button)
