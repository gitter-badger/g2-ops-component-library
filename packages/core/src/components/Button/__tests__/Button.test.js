import React from 'react'
import renderer from 'react-test-renderer'

import { wrapWithMaterialUIContext } from 'utilities/wrapWithContext'

import {Button} from '../Button'

// This is also a wrapper around MaterialUI. Added snapshots, to track how these things change over time
describe('<Button />', () => {
  test('should render properly', () => {
    const buttonTypes = [ 'primary', 'secondary', 'inactive', 'add', 'edit', 'delete' ]
    buttonTypes.forEach((type) => {
      const tree = renderer.create(wrapWithMaterialUIContext(<Button type={type} />))
      expect(tree).toMatchSnapshot()
    })
  })
})
