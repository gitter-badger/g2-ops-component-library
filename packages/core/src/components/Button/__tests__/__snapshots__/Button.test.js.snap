// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Button /> should render properly 1`] = `
<div
  className="copartButton primaryButton "
  style={
    Object {
      "WebkitTapHighlightColor": "rgba(0,0,0,0)",
      "backgroundColor": "#ffffff",
      "borderRadius": 2,
      "boxShadow": "0 1px 6px rgba(0, 0, 0, 0.12),
         0 1px 4px rgba(0, 0, 0, 0.12)",
      "boxSizing": "border-box",
      "color": "rgba(0, 0, 0, 0.87)",
      "display": "inline-block",
      "fontFamily": "Roboto, sans-serif",
      "minWidth": 88,
      "muiPrepared": true,
      "transition": "all 450ms cubic-bezier(0.23, 1, 0.32, 1) 0ms",
    }
  }
>
  <button
    disabled={false}
    onBlur={[Function]}
    onClick={[Function]}
    onFocus={[Function]}
    onKeyDown={[Function]}
    onKeyUp={[Function]}
    onMouseDown={[Function]}
    onMouseEnter={[Function]}
    onMouseLeave={[Function]}
    onMouseUp={[Function]}
    onTouchEnd={[Function]}
    onTouchStart={[Function]}
    style={
      Object {
        "WebkitTapHighlightColor": "rgba(0, 0, 0, 0)",
        "backgroundColor": "#ffffff",
        "border": 10,
        "borderRadius": 2,
        "boxSizing": "border-box",
        "cursor": "pointer",
        "display": "inline-block",
        "fontFamily": "Roboto, sans-serif",
        "fontSize": "inherit",
        "fontWeight": "inherit",
        "height": 36,
        "lineHeight": "36px",
        "margin": 0,
        "muiPrepared": true,
        "outline": "none",
        "padding": 0,
        "position": "relative",
        "textAlign": "center",
        "textDecoration": "none",
        "transition": "all 450ms cubic-bezier(0.23, 1, 0.32, 1) 0ms",
        "verticalAlign": null,
        "width": "100%",
      }
    }
    tabIndex={0}
    type="button"
  >
    <div
      onMouseDown={[Function]}
      onMouseLeave={[Function]}
      onMouseUp={[Function]}
      onTouchEnd={[Function]}
      onTouchStart={[Function]}
    >
      <div
        style={
          Object {
            "backgroundColor": false,
            "borderRadius": 2,
            "height": 36,
            "muiPrepared": true,
            "top": 0,
            "transition": "all 450ms cubic-bezier(0.23, 1, 0.32, 1) 0ms",
          }
        }
      />
    </div>
  </button>
</div>
`;

exports[`<Button /> should render properly 2`] = `
<div
  className="copartButton secondaryButton "
  style={
    Object {
      "WebkitTapHighlightColor": "rgba(0,0,0,0)",
      "backgroundColor": "#ffffff",
      "borderRadius": 2,
      "boxShadow": "0 1px 6px rgba(0, 0, 0, 0.12),
         0 1px 4px rgba(0, 0, 0, 0.12)",
      "boxSizing": "border-box",
      "color": "rgba(0, 0, 0, 0.87)",
      "display": "inline-block",
      "fontFamily": "Roboto, sans-serif",
      "minWidth": 88,
      "muiPrepared": true,
      "transition": "all 450ms cubic-bezier(0.23, 1, 0.32, 1) 0ms",
    }
  }
>
  <button
    disabled={false}
    onBlur={[Function]}
    onClick={[Function]}
    onFocus={[Function]}
    onKeyDown={[Function]}
    onKeyUp={[Function]}
    onMouseDown={[Function]}
    onMouseEnter={[Function]}
    onMouseLeave={[Function]}
    onMouseUp={[Function]}
    onTouchEnd={[Function]}
    onTouchStart={[Function]}
    style={
      Object {
        "WebkitTapHighlightColor": "rgba(0, 0, 0, 0)",
        "backgroundColor": "#ffffff",
        "border": 10,
        "borderRadius": 2,
        "boxSizing": "border-box",
        "cursor": "pointer",
        "display": "inline-block",
        "fontFamily": "Roboto, sans-serif",
        "fontSize": "inherit",
        "fontWeight": "inherit",
        "height": 36,
        "lineHeight": "36px",
        "margin": 0,
        "muiPrepared": true,
        "outline": "none",
        "padding": 0,
        "position": "relative",
        "textAlign": "center",
        "textDecoration": "none",
        "transition": "all 450ms cubic-bezier(0.23, 1, 0.32, 1) 0ms",
        "verticalAlign": null,
        "width": "100%",
      }
    }
    tabIndex={0}
    type="button"
  >
    <div
      onMouseDown={[Function]}
      onMouseLeave={[Function]}
      onMouseUp={[Function]}
      onTouchEnd={[Function]}
      onTouchStart={[Function]}
    >
      <div
        style={
          Object {
            "backgroundColor": false,
            "borderRadius": 2,
            "height": 36,
            "muiPrepared": true,
            "top": 0,
            "transition": "all 450ms cubic-bezier(0.23, 1, 0.32, 1) 0ms",
          }
        }
      />
    </div>
  </button>
</div>
`;

exports[`<Button /> should render properly 3`] = `
<div
  className="copartButton inactiveButton "
  style={
    Object {
      "WebkitTapHighlightColor": "rgba(0,0,0,0)",
      "backgroundColor": "#ffffff",
      "borderRadius": 2,
      "boxShadow": "0 1px 6px rgba(0, 0, 0, 0.12),
         0 1px 4px rgba(0, 0, 0, 0.12)",
      "boxSizing": "border-box",
      "color": "rgba(0, 0, 0, 0.87)",
      "display": "inline-block",
      "fontFamily": "Roboto, sans-serif",
      "minWidth": 88,
      "muiPrepared": true,
      "transition": "all 450ms cubic-bezier(0.23, 1, 0.32, 1) 0ms",
    }
  }
>
  <button
    disabled={false}
    onBlur={[Function]}
    onClick={[Function]}
    onFocus={[Function]}
    onKeyDown={[Function]}
    onKeyUp={[Function]}
    onMouseDown={[Function]}
    onMouseEnter={[Function]}
    onMouseLeave={[Function]}
    onMouseUp={[Function]}
    onTouchEnd={[Function]}
    onTouchStart={[Function]}
    style={
      Object {
        "WebkitTapHighlightColor": "rgba(0, 0, 0, 0)",
        "backgroundColor": "#ffffff",
        "border": 10,
        "borderRadius": 2,
        "boxSizing": "border-box",
        "cursor": "pointer",
        "display": "inline-block",
        "fontFamily": "Roboto, sans-serif",
        "fontSize": "inherit",
        "fontWeight": "inherit",
        "height": 36,
        "lineHeight": "36px",
        "margin": 0,
        "muiPrepared": true,
        "outline": "none",
        "padding": 0,
        "position": "relative",
        "textAlign": "center",
        "textDecoration": "none",
        "transition": "all 450ms cubic-bezier(0.23, 1, 0.32, 1) 0ms",
        "verticalAlign": null,
        "width": "100%",
      }
    }
    tabIndex={0}
    type="button"
  >
    <div
      onMouseDown={[Function]}
      onMouseLeave={[Function]}
      onMouseUp={[Function]}
      onTouchEnd={[Function]}
      onTouchStart={[Function]}
    >
      <div
        style={
          Object {
            "backgroundColor": false,
            "borderRadius": 2,
            "height": 36,
            "muiPrepared": true,
            "top": 0,
            "transition": "all 450ms cubic-bezier(0.23, 1, 0.32, 1) 0ms",
          }
        }
      />
    </div>
  </button>
</div>
`;

exports[`<Button /> should render properly 4`] = `
<div
  className="copartButton primaryButton addButton "
  style={
    Object {
      "WebkitTapHighlightColor": "rgba(0,0,0,0)",
      "backgroundColor": "#ffffff",
      "borderRadius": 2,
      "boxShadow": "0 1px 6px rgba(0, 0, 0, 0.12),
         0 1px 4px rgba(0, 0, 0, 0.12)",
      "boxSizing": "border-box",
      "color": "rgba(0, 0, 0, 0.87)",
      "display": "inline-block",
      "fontFamily": "Roboto, sans-serif",
      "minWidth": 88,
      "muiPrepared": true,
      "transition": "all 450ms cubic-bezier(0.23, 1, 0.32, 1) 0ms",
    }
  }
>
  <button
    disabled={false}
    onBlur={[Function]}
    onClick={[Function]}
    onFocus={[Function]}
    onKeyDown={[Function]}
    onKeyUp={[Function]}
    onMouseDown={[Function]}
    onMouseEnter={[Function]}
    onMouseLeave={[Function]}
    onMouseUp={[Function]}
    onTouchEnd={[Function]}
    onTouchStart={[Function]}
    style={
      Object {
        "WebkitTapHighlightColor": "rgba(0, 0, 0, 0)",
        "backgroundColor": "#ffffff",
        "border": 10,
        "borderRadius": 2,
        "boxSizing": "border-box",
        "cursor": "pointer",
        "display": "inline-block",
        "fontFamily": "Roboto, sans-serif",
        "fontSize": "inherit",
        "fontWeight": "inherit",
        "height": 36,
        "lineHeight": "36px",
        "margin": 0,
        "muiPrepared": true,
        "outline": "none",
        "padding": 0,
        "position": "relative",
        "textAlign": "center",
        "textDecoration": "none",
        "transition": "all 450ms cubic-bezier(0.23, 1, 0.32, 1) 0ms",
        "verticalAlign": null,
        "width": "100%",
      }
    }
    tabIndex={0}
    type="button"
  >
    <div
      onMouseDown={[Function]}
      onMouseLeave={[Function]}
      onMouseUp={[Function]}
      onTouchEnd={[Function]}
      onTouchStart={[Function]}
    >
      <div
        style={
          Object {
            "backgroundColor": false,
            "borderRadius": 2,
            "height": 36,
            "muiPrepared": true,
            "top": 0,
            "transition": "all 450ms cubic-bezier(0.23, 1, 0.32, 1) 0ms",
          }
        }
      >
        <svg
          onMouseEnter={[Function]}
          onMouseLeave={[Function]}
          style={
            Object {
              "color": "rgba(0, 0, 0, 0.87)",
              "display": "inline-block",
              "fill": "rgba(0, 0, 0, 0.87)",
              "height": 24,
              "marginLeft": 12,
              "marginRight": 0,
              "muiPrepared": true,
              "transition": "all 450ms cubic-bezier(0.23, 1, 0.32, 1) 0ms",
              "userSelect": "none",
              "verticalAlign": "middle",
              "width": 24,
            }
          }
          viewBox="0 0 24 24"
        >
          <path
            d="M19 13h-6v6h-2v-6H5v-2h6V5h2v6h6v2z"
          />
        </svg>
        <span
          style={
            Object {
              "color": "rgba(0, 0, 0, 0.87)",
              "fontSize": 14,
              "fontWeight": 500,
              "letterSpacing": 0,
              "margin": 0,
              "muiPrepared": true,
              "opacity": 1,
              "paddingLeft": 8,
              "paddingRight": 16,
              "position": "relative",
              "textTransform": "uppercase",
              "userSelect": "none",
            }
          }
        >
          Add
        </span>
      </div>
    </div>
  </button>
</div>
`;

exports[`<Button /> should render properly 5`] = `
<div
  className="copartButton primaryButton editButton "
  style={
    Object {
      "WebkitTapHighlightColor": "rgba(0,0,0,0)",
      "backgroundColor": "#ffffff",
      "borderRadius": 2,
      "boxShadow": "0 1px 6px rgba(0, 0, 0, 0.12),
         0 1px 4px rgba(0, 0, 0, 0.12)",
      "boxSizing": "border-box",
      "color": "rgba(0, 0, 0, 0.87)",
      "display": "inline-block",
      "fontFamily": "Roboto, sans-serif",
      "minWidth": 88,
      "muiPrepared": true,
      "transition": "all 450ms cubic-bezier(0.23, 1, 0.32, 1) 0ms",
    }
  }
>
  <button
    disabled={false}
    onBlur={[Function]}
    onClick={[Function]}
    onFocus={[Function]}
    onKeyDown={[Function]}
    onKeyUp={[Function]}
    onMouseDown={[Function]}
    onMouseEnter={[Function]}
    onMouseLeave={[Function]}
    onMouseUp={[Function]}
    onTouchEnd={[Function]}
    onTouchStart={[Function]}
    style={
      Object {
        "WebkitTapHighlightColor": "rgba(0, 0, 0, 0)",
        "backgroundColor": "#ffffff",
        "border": 10,
        "borderRadius": 2,
        "boxSizing": "border-box",
        "cursor": "pointer",
        "display": "inline-block",
        "fontFamily": "Roboto, sans-serif",
        "fontSize": "inherit",
        "fontWeight": "inherit",
        "height": 36,
        "lineHeight": "36px",
        "margin": 0,
        "muiPrepared": true,
        "outline": "none",
        "padding": 0,
        "position": "relative",
        "textAlign": "center",
        "textDecoration": "none",
        "transition": "all 450ms cubic-bezier(0.23, 1, 0.32, 1) 0ms",
        "verticalAlign": null,
        "width": "100%",
      }
    }
    tabIndex={0}
    type="button"
  >
    <div
      onMouseDown={[Function]}
      onMouseLeave={[Function]}
      onMouseUp={[Function]}
      onTouchEnd={[Function]}
      onTouchStart={[Function]}
    >
      <div
        style={
          Object {
            "backgroundColor": false,
            "borderRadius": 2,
            "height": 36,
            "muiPrepared": true,
            "top": 0,
            "transition": "all 450ms cubic-bezier(0.23, 1, 0.32, 1) 0ms",
          }
        }
      >
        <svg
          onMouseEnter={[Function]}
          onMouseLeave={[Function]}
          style={
            Object {
              "color": "rgba(0, 0, 0, 0.87)",
              "display": "inline-block",
              "fill": "rgba(0, 0, 0, 0.87)",
              "height": 24,
              "marginLeft": 12,
              "marginRight": 0,
              "muiPrepared": true,
              "transition": "all 450ms cubic-bezier(0.23, 1, 0.32, 1) 0ms",
              "userSelect": "none",
              "verticalAlign": "middle",
              "width": 24,
            }
          }
          viewBox="0 0 24 24"
        >
          <path
            d="M3 17.25V21h3.75L17.81 9.94l-3.75-3.75L3 17.25zM20.71 7.04c.39-.39.39-1.02 0-1.41l-2.34-2.34c-.39-.39-1.02-.39-1.41 0l-1.83 1.83 3.75 3.75 1.83-1.83z"
          />
        </svg>
        <span
          style={
            Object {
              "color": "rgba(0, 0, 0, 0.87)",
              "fontSize": 14,
              "fontWeight": 500,
              "letterSpacing": 0,
              "margin": 0,
              "muiPrepared": true,
              "opacity": 1,
              "paddingLeft": 8,
              "paddingRight": 16,
              "position": "relative",
              "textTransform": "uppercase",
              "userSelect": "none",
            }
          }
        >
          Edit
        </span>
      </div>
    </div>
  </button>
</div>
`;

exports[`<Button /> should render properly 6`] = `
<div
  className="copartButton deleteButton "
  style={
    Object {
      "WebkitTapHighlightColor": "rgba(0,0,0,0)",
      "backgroundColor": "#ffffff",
      "borderRadius": 2,
      "boxShadow": "0 1px 6px rgba(0, 0, 0, 0.12),
         0 1px 4px rgba(0, 0, 0, 0.12)",
      "boxSizing": "border-box",
      "color": "rgba(0, 0, 0, 0.87)",
      "display": "inline-block",
      "fontFamily": "Roboto, sans-serif",
      "minWidth": 88,
      "muiPrepared": true,
      "transition": "all 450ms cubic-bezier(0.23, 1, 0.32, 1) 0ms",
    }
  }
>
  <button
    disabled={false}
    onBlur={[Function]}
    onClick={[Function]}
    onFocus={[Function]}
    onKeyDown={[Function]}
    onKeyUp={[Function]}
    onMouseDown={[Function]}
    onMouseEnter={[Function]}
    onMouseLeave={[Function]}
    onMouseUp={[Function]}
    onTouchEnd={[Function]}
    onTouchStart={[Function]}
    style={
      Object {
        "WebkitTapHighlightColor": "rgba(0, 0, 0, 0)",
        "backgroundColor": "#ffffff",
        "border": 10,
        "borderRadius": 2,
        "boxSizing": "border-box",
        "cursor": "pointer",
        "display": "inline-block",
        "fontFamily": "Roboto, sans-serif",
        "fontSize": "inherit",
        "fontWeight": "inherit",
        "height": 36,
        "lineHeight": "36px",
        "margin": 0,
        "muiPrepared": true,
        "outline": "none",
        "padding": 0,
        "position": "relative",
        "textAlign": "center",
        "textDecoration": "none",
        "transition": "all 450ms cubic-bezier(0.23, 1, 0.32, 1) 0ms",
        "verticalAlign": null,
        "width": "100%",
      }
    }
    tabIndex={0}
    type="button"
  >
    <div
      onMouseDown={[Function]}
      onMouseLeave={[Function]}
      onMouseUp={[Function]}
      onTouchEnd={[Function]}
      onTouchStart={[Function]}
    >
      <div
        style={
          Object {
            "backgroundColor": false,
            "borderRadius": 2,
            "height": 36,
            "muiPrepared": true,
            "top": 0,
            "transition": "all 450ms cubic-bezier(0.23, 1, 0.32, 1) 0ms",
          }
        }
      >
        <svg
          onMouseEnter={[Function]}
          onMouseLeave={[Function]}
          style={
            Object {
              "color": "rgba(0, 0, 0, 0.87)",
              "display": "inline-block",
              "fill": "rgba(0, 0, 0, 0.87)",
              "height": 24,
              "marginLeft": 12,
              "marginRight": 0,
              "muiPrepared": true,
              "transition": "all 450ms cubic-bezier(0.23, 1, 0.32, 1) 0ms",
              "userSelect": "none",
              "verticalAlign": "middle",
              "width": 24,
            }
          }
          viewBox="0 0 24 24"
        >
          <path
            d="M6 19c0 1.1.9 2 2 2h8c1.1 0 2-.9 2-2V7H6v12zm2.46-7.12l1.41-1.41L12 12.59l2.12-2.12 1.41 1.41L13.41 14l2.12 2.12-1.41 1.41L12 15.41l-2.12 2.12-1.41-1.41L10.59 14l-2.13-2.12zM15.5 4l-1-1h-5l-1 1H5v2h14V4z"
          />
        </svg>
        <span
          style={
            Object {
              "color": "rgba(0, 0, 0, 0.87)",
              "fontSize": 14,
              "fontWeight": 500,
              "letterSpacing": 0,
              "margin": 0,
              "muiPrepared": true,
              "opacity": 1,
              "paddingLeft": 8,
              "paddingRight": 16,
              "position": "relative",
              "textTransform": "uppercase",
              "userSelect": "none",
            }
          }
        >
          Delete
        </span>
      </div>
    </div>
  </button>
</div>
`;
